src/main/webapp/app/tutor-exercise-dashboard/tutor-exercise-dashboard.component.ts

```diff
@@ -165,77 +168,62 @@ export class TutorExerciseDashboardComponent implements OnInit {
         );
     }
 
-    // TODO CZ: too much duplicated code
     private getSubmissions(): void {
+        let submissionsObservable: Observable<HttpResponse<Submission[]>>;
```

Comment:
Good use of observables!

If it is neither a text or modeling exercise, the observable will be undefined (pipe of undefined is not a function).
I think it would better to use an empty observable (`of()`) just to make sure no console error is thrown.
(This will behave like the previous implementation).

If you want to go 100% functional, you could also use the [iif operator](https://www.learnrxjs.io/operators/conditional/iif.html).

Reply:
Thanks, you are right. 

I tried to use the `iif` operator like this: `iif(() => !!submissionsObservable, submissionsObservable).pipe(...` but I then got the error message `Variable 'submissionsObservable' is used before being assigned`. I don't know if I did something wrong. In the end I went with the `of()`.